packages/compass-components/src/components/interactive-popover.tsx

```diff
@@ -132,6 +143,19 @@ function InteractivePopover({
     [onClose]
   );
 
+  // Delay activating the focus trap until the popover has mounted.
+  useEffect(() => {
+    const activateFocusTrapTimeout = open
+      ? setTimeout(() => setIsFocusTrapActive(true), popoverDelayMS)
```

Comment:
this is a bit weird .. did you stumble on an instance where this was necessary? it should work well also without, the fallback as far as I understand would be hit only in JSDOM

Reply:
Removed it and instead we're now using the `checkCanFocusTrap` in the `focusTrapOptions`. We do an immediate for the popover to mount and render so that the focus trap can then find the focusable elements. 
https://github.com/focus-trap/focus-trap#createoptions
Here's an example of using it with shadow dom: https://github.com/focus-trap/focus-trap-react/blob/master/demo/js/demo-with-shadow-dom.js
https://github.com/focus-trap/focus-trap-react#focustrapoptions
From my understanding of the docs I don't think we want to use the `fallbackFocus` argument as we already are supplying the button which is tabbable in the dom.

I think the focus trap doesn't work nicely with more complex unit tests as a result of this:
https://github.com/focus-trap/focus-trap-react#testing-in-jsdom
So I think the end to end tests around query history end up being useful for testing more of this functionality.

Reply:
From what we have tried out the trap had no issues "IRL" outside of tests, but it did make tests fail and `fallbackFocus` is what folks suggest for jsdom, is also what LG is doing in modals  🤷‍♂️.

> So I think the end to end tests around query history end up being useful for testing more of this functionality.

But if we want we could write unit tests, or would they fail? If they still fail we may keep investigating for something else (even disabling the focus trap if necessary), as we don't want to be restricted on tests everywhere we use the `InteractivePopover`.

Reply:
`fallbackFocus` works good, updated. Better than the other way. Thanks for the suggestion